core/src/main/java/com/oath/oak/Chunk.java

```diff
@@ -1156,10 +1175,28 @@ private void findNewNextInStack() {
             }
         }
 
+        private void pushToStack(boolean compareWithPrevAnchor) {
```

Comment:
please add documentation (explaining the steps)
it is difficult to reason about why this code is correct